feat(workflows): [INFRA-496] add Helm chart support to artifacts-cicd (#221)
aerospike/shared-workflows: .github/workflows/artifacts-cicd/README.md

Author: Not in this pr but some documentation debt that we should cover here since i am noticing it. At the beginning we are still using the language that this is superior to the composable workflow. It isn't it is just a simple alternative that is less flexible but simpler.

Author: On second look you're right, this is in scope. Fixed in 4339c6e: reframed both the artifacts-cicd README "Design Philosophy" section and the project-level CLAUDE.md "Core Workflows" section. Composable is now described as a first-class peer path rather than "_Internal._ implementation detail" or "smell" to use directly.

| Helm    | `helm-dev-local`    | `version`, `package_name`, `helm.name`, `helm.version`                             |
| Generic | `generic-dev-local` | `version`, `package_name`                                                          |

Companion files (`.asc` signatures, `.pom` files, helm `.prov` provenance) are automatically gathered alongside their parent artifacts. Helm registers `.prov` (helm-native provenance signature) as its companion; the orphan `.tgz.asc` produced by GPG sign-artifacts is intentionally not gathered or uploaded for helm charts because `.prov` is the canonical chart signature.

### Helm charts

Helm charts publish to a JFrog classic Helm repository (`{project}-helm-dev-local`, a `artifactory_local_helm_repository` resource in tf-artifactory). JFrog auto-generates `index.yaml` from uploaded `.tgz` files, so consumers add the repo and install with the standard helm CLI:

```bash
helm repo add aerospike-helm \
  https://artifact.aerospike.io/artifactory/api/helm/{project}-helm-dev-local
helm repo update
helm install my-release aerospike-helm/<chart-name>
```

> Packaging is the consumer's build-script responsibility, mirroring the npm/pypi/go convention. Run `helm package charts/<name> -d <output-dir>` and the deploy stage auto-detects the resulting `.tgz` (sniffed via `<chart>/Chart.yaml` containing `apiVersion`, `name`, and `version`).

Author: ```suggestion
Packaging is the consumer's build-script responsibility, mirroring the npm/pypi/go convention. Run `helm package charts/<name> -d <output-dir>` and the deploy stage auto-detects the resulting `.tgz`.
```

Author: Applied in fe1f628.

Signing happens automatically in the sign stage. `sign-artifacts` recognises chart `.tgz` files and produces a helm-native `.prov` (GPG-clearsigned Chart.yaml plus a sha256 of the tarball) using the same GPG key it uses for deb/rpm/`.asc`. Build-scripts should run plain `helm package` (no `--sign`) and let the workflow handle provenance. The `.prov` rides alongside the chart through deploy and lands as a companion in the helm repo. Consumers verify with `helm install --verify` or `helm verify chart.tgz`.

> Chart linting and unit testing are out of scope for this orchestrator (publishing is a release-time concern, linting is a PR-time concern). Run [chart-testing (`ct`)](https://github.com/helm/chart-testing) via the [helm/chart-testing-action](https://github.com/helm/chart-testing-action) as a separate `pull_request`-triggered workflow. The [aerospike/helm-aerospike-vector-search](https://github.com/aerospike/helm-aerospike-vector-search) repo has a working `lint.yaml` + `.ct.yaml` you can copy verbatim.

Author: ```suggestion
Chart linting and unit testing are the user's responsibility. Suggestion is to run [chart-testing (`ct`)](https://github.com/helm/chart-testing) via the [helm/chart-testing-action](https://github.com/helm/chart-testing-action) as a separate `pull_request`-triggered workflow. 
```

Author: Applied in fe1f628.
